Assets/Source/Components/NavigationMesh/NavigationMeshComponent.cs

```diff
@@ -0,0 +1,135 @@
+﻿using Assets.Source.AStar;
+using Assets.Source.Components.Base;
+using System.Linq;
+using UnityEngine;
+
+namespace Assets.Source.Components.NavigationMesh
+{
+    public class NavigationMeshComponent : ComponentBase
+    {
+        [Tooltip("How many tiles going horizontally")]
+        [SerializeField]
+        private int gridWidth = 20;
+        public int GridWidth => gridWidth;
+
+        [Tooltip("How many tiles going vertically")]
+        [SerializeField]
+        private int gridHeight = 15;
+        public int GridHeight => gridHeight;
+
+
+        [Tooltip("The size of each tile in units.  Should be large enough for a pathfinder to pass through " +
+            "without clipping the edge of corners.")]
+        [SerializeField]
+        private float tileSize = 0.5f;
+
+        [Tooltip("Recalculates the entire navigation mesh every frame.  Leave this on if things in your " +
+            "level move around, or set to false if the level is static for a performance boost.")]
+        [SerializeField]
+        private bool syncEveryFrame = true;
+
+        private Node[][] nodes;
+
+        public override void ComponentAwake()
+        {
+            nodes = new Node[gridWidth][];
```

Comment:
why not just put gridHeight in here initially instead of on line 45?

Reply:
because we're using a 2d array, not a jagged array, so each array in the y dimension could theoretically be a different size

Reply:
(even though its not rn)